src/CommandLine/CommandLineParser.cpp

```diff
@@ -576,50 +593,59 @@ bool CommandLineParser::parseCommandLine(int argc, const char** argv) {
 
   std::vector<std::string> cmd_line;
   for (int i = 1; i < argc; i++) {
-    std::string arg = StringUtils::unquoted(argv[i]);
-    cmd_line.emplace_back(StringUtils::trim(arg));
-    if (arg == "-cd") {
-      std::string newDir = StringUtils::unquoted(argv[i + 1]);
-      StringUtils::trim(newDir);
-      int ret = chdir(newDir.c_str());
-      if (ret < 0) {
-        std::cerr << "Could not change directory to " << newDir << "\n"
-                  << std::endl;
+    cmd_line.emplace_back(undecorateArg(argv[i]));
+    const std::string& arg = cmd_line.back();
+
+    if (arg == "-help" || arg == "-h" || arg == "--help") {
+      m_help = true;
+      std::string help = printStringArray(helpText);
+      m_errors->init();
+      logBanner(argc, argv);
+      std::cout << help;
+      return true;
+    }
+    if (arg == "--version") {
+      std::cout << BuildIdentifier() << std::flush;
+      m_help = true;
+      return true;
+    } else if (arg == "-cd") {
+      if (i < argc - 1) {
+        std::string newDir(undecorateArg(argv[i + 1]));
+        int ret = chdir(newDir.c_str());
+        if (ret < 0) {
+          std::cerr << "Could not change directory to " << newDir << std::endl;
+        }
       }
     } else if (arg == "-builtin") {
       if (i < argc - 1) {
-        arg = StringUtils::unquoted(argv[i + 1]);
-        m_builtinPath = StringUtils::trim(arg);
+        m_builtinPath = undecorateArg(argv[i + 1]);
       }
     } else if (arg == "-l") {
       if (i < argc - 1) {
-        arg = StringUtils::unquoted(argv[i + 1]);
-        m_logFileId = m_symbolTable->registerSymbol(StringUtils::trim(arg));
-        m_logFileSpecified = true;
+        m_logFileId = m_symbolTable->registerSymbol(undecorateArg(argv[i + 1]));
       }
-    } else if (arg.find("-D") != std::string::npos) {
+    } else if (arg.find("-D") == 0) {
```

Comment:
-DFOO    
      and 
-DBAR=BLAH  

are both valid syntax, does your code work for both?

Reply:
Yes, this check is to ensure that the '-D' is at the start of the string and not somewhere in the middle. So, something like "-XABC-DEF" shouldn't succeed.